com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -2109,8 +2142,16 @@ def partition(self, attribute):
         Returns:
             Window: Window that will be triggered.
         """
+
+        if 'partitioned' in self._config and self._config['partitioned']:
```

Comment:
Since a new window is returned is this check needed? The partitioning replaces any existing partitioning.

Reply:
I think it would be unexpected for the user for multiple calls to partition to be accepted, but only one of them to be applied.  I think if someone calls partition multiple times, it is probably either an error, or the user expects all of the calls to apply.  Failing with an error message helps the user understand what will be applied in these cases.

Reply:
Each one would be applied, each partition call returns a new window. That needs to be clear from the docstring. See `trigger`.